File: examples/spring-boot/src/main/java/com/example/koma/service/KomaService.java
package com.example.koma.service;

import com.example.koma.config.KomaProperties;
import java.util.Map;
import org.springframework.stereotype.Service;

@Service
public class KomaService {
  private final KomaProperties props;

  public KomaService(KomaProperties props) {
    this.props = props;
  }

  public Map<String, Object> createQrSession(String amount, String currency, String productId) {
    return Map.of(
        "amount", amount,
        "currency", currency,
        "productId", productId,
        "merchantId", props.merchantId()
    );
  }
}

File: examples/php/src/KomaService.php
<?php

final class KomaService
{
    public function __construct(private array $config) {}

    public function createQrSession(string $amount, string $currency, string $productId): array
    {
        return [
            'amount' => $amount,
            'currency' => $currency,
            'productId' => $productId,
            'merchantId' => $this->config['merchant_id'],
        ];
    }
}
